### PR TITLE
Mvp

### PR DESCRIPTION
**Pull request by eloiseyaz**

### CODE REVIEW

File: src/main/scala/Menu/MenuItem.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package Menu
+
+abstract class MenuItem(val name: String, val itemType: MenuItem.ItemType, val price: Double, val premium: Boolean) {
+  val serviceCharge: Double
```
Comment (waris1hezb) [inline src/main/scala/Menu/MenuItem.scala]: Good use of abstract class and well organised file management. 

Maybe think about having all the logic of the service charge in a common method (e.g here in the MenuItem class) to avoid duplication. Service charge calculation is repeated in each subclass which still works. You can just create def ServiceCharge: and use if/else and pattern matching to handle all cases.

File: src/main/scala/CafeLogic.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  val coke = Drink.apply("Coca Cola", 2)
+  val cake = ColdFood.apply("Cake", 4, true)
+  val newOrder = Order.apply(List((crisps, 2), (coke, 1)))
+  val regMenu: List[MenuItem] = List(nuggets, crisps, coke)
+
+  //  def placeOrder(order: Order): Either[String, Order] = {
+  //  }
+
+  def getBill(order: Order, serviceCharge: Option[Double] = None): List[(String, Int, Double)] = {
+    val sCharge = serviceCharge.getOrElse(order.serviceCharge)
```
Comment (waris1hezb) [inline src/main/scala/CafeLogic.scala]: Maybe use a more descriptive name for variable to make it clearer. e.g serviceChargeMultiplier for  sCharge.

File: src/main/scala/CafeLogic.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  val crisps = ColdFood.apply("Crisps", 1.5)
+  val coke = Drink.apply("Coca Cola", 2)
+  val cake = ColdFood.apply("Cake", 4, true)
+  val newOrder = Order.apply(List((crisps, 2), (coke, 1)))
+  val regMenu: List[MenuItem] = List(nuggets, crisps, coke)
+
+  //  def placeOrder(order: Order): Either[String, Order] = {
+  //  }
+
+  def getBill(order: Order, serviceCharge: Option[Double] = None): List[(String, Int, Double)] = {
```
Comment (waris1hezb) [inline src/main/scala/CafeLogic.scala]: You could use Eithers to handle error cases. maybe something like Either[String, List[(String, Int, Double)]] . The string can print an error message 
e.g if order is empty or typo

File: src/main/scala/CafeLogic.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    val sCharge = serviceCharge.getOrElse(order.serviceCharge)
+    order.order.map(item => (item._1.name, item._2, (item._1.price * item._2 * sCharge * 100).round.toDouble / 100))
+  }
+
+  def getTotalCost(order: Order, serviceCharge: Option[Double] = None): Double = {
```
Comment (waris1hezb) [inline src/main/scala/CafeLogic.scala:28]: Could use Either here too e.g Either[String, Double]. Great job on the logic overall

File: src/main/scala/Menu/MenuItem.scala [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+object MenuItem {
+  sealed trait ItemType
+  case object ColdFood extends ItemType
+  case object HotFood extends ItemType
+  case object Drink extends ItemType
+}
```
Comment (waris1hezb) [inline src/main/scala/Menu/MenuItem.scala:21]: You're OOP is great.  Love that you've split everything up and made it modular.
One potential improvement could be to use case objects for MenuItem names to ensure type safety and consistency. Have it all in a sealed trait then create case objects
e.g case object Crisps extends MenuItemName

THIS IS VERY EXTRA BTW and you don't need to do it. But, you could create sealed traits for foods and drinks that extend name , and then you can create case objects for everything and they can all extend their specific categories. 

This approach ensure that only valid names can be used, reducing the risk of errors. 
You can ask me if you want more clarification. I repeat, this is very extra and long but thought it would be nice to put the idea in your head. Food for thought

File: src/main/scala/CafeLogic.scala [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    (order.cost * serviceCharge.getOrElse(order.serviceCharge) * 100).round.toDouble / 100
+  }
+
```
Comment (waris1hezb) [inline src/main/scala/CafeLogic.scala:31]: Maybe create method to factor in stock of items and update stock. You can update you classes to have stock as a parameter or maybe create an inventory class/method

File: src/main/scala/Order.scala [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import Menu.MenuItem
+
```
Comment (waris1hezb) [inline src/main/scala/Order.scala:2]: what happens if order quantity is negative?
Maybe can check stock here?

File: src/test/scala/CafeLogicSpec.scala [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+      "the service charge is specified" in {
+        val order: Order = Order.apply(List((smoothie, 1), (coke, 1), (crisps, 1)))
+        val expected: List[(String, Int, Double)] = List(("Berry Smoothie", 1, 3.45), ("Coca Cola", 1, 2.3), ("Crisps", 1, 1.73))
+        getBill(order, Some(1.15))
+
+      }
+
+    }
+
```
Comment (waris1hezb) [inline src/test/scala/CafeLogicSpec.scala:53]: Maybe add tests for edge cases such as empty orders, negative quantities, invalid service charges

File: src/test/scala/OrderSpec.scala [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+      "there is none of an item" in {
+        val emptyOrder = Order.apply(List((crisps, 0), (nuggets, 0), (coke, 0)))
+        emptyOrder.cost shouldBe 0
+        val someEmptyOrder = Order.apply(List((crisps, 1), (nuggets, 0), (coke, 3)))
+        someEmptyOrder.cost shouldBe 7.5
+      }
+
+    }
+
+  }
```
Comment (waris1hezb) [inline src/test/scala/OrderSpec.scala:118]: Overall good job, its an amazing project and you should be proud of yourself. Well done!!! 
The code is generally clear and well-structured. Class and method names are descriptive, and the logic is easy to follow. Testing covered most cases.

File: src/main/scala/CafeLogic.scala [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  val cake = ColdFood.apply("Cake", 4, true)
+  val newOrder = Order.apply(List((crisps, 2), (coke, 1)))
+  val regMenu: List[MenuItem] = List(nuggets, crisps, coke)
+
+  //  def placeOrder(order: Order): Either[String, Order] = {
+  //  }
+
+  def getBill(order: Order, serviceCharge: Option[Double] = None): List[(String, Int, Double)] = {
+    val sCharge = serviceCharge.getOrElse(order.serviceCharge)
+    order.order.map(item => (item._1.name, item._2, (item._1.price * item._2 * sCharge * 100).round.toDouble / 100))
```
Comment (waris1hezb) [inline src/main/scala/CafeLogic.scala]: two orders from different places. Maybe change one of the names to make it less confusing
